django/db/models/enums.py

```diff
@@ -0,0 +1,76 @@
+import enum
+
+from django.utils.functional import Promise
+
+__all__ = ['ChoiceIntEnum', 'ChoiceStrEnum']
+
+
+class ChoiceEnumMeta(enum.EnumMeta):
+    """A metaclass for creating a enum choices."""
+
+    def __new__(metacls, classname, bases, classdict):
+        labels = []
+        for key in classdict._member_names:
+            value = classdict[key]
+            if (
+                isinstance(value, (list, tuple)) and
+                len(value) > 1 and
+                isinstance(value[-1], (Promise, str))
+            ):
+                try:
+                    *value, label = value
+                except ValueError as e:
+                    raise ValueError('Invalid member for %s: %s.' % (classname, key)) from e
+                else:
+                    value = tuple(value)
+            else:
+                label = key.replace('_', ' ').title()
+            labels.append(label)
+            # Use dict.__setitem__() to suppress defenses against double
+            # assignment in enum's classdict.
+            dict.__setitem__(classdict, key, value)
+        cls = super().__new__(metacls, classname, bases, classdict)
+        cls._value2label_map_ = dict(zip(cls._value2member_map_, labels))
+        # Add a 'label' property to instances of enum which uses the enum
+        # member that is passed in as ``self`` as the value to use when looking
+        # up the label in the choices.
+        cls.label = property(lambda self: cls._value2label_map_.get(self.value))
+        return enum.unique(cls)
+
+    def __contains__(cls, member):
+        if not isinstance(member, enum.Enum):
+            # Allow non-enums to match against member values.
+            return member in {x.value for x in cls}
+        return super().__contains__(member)
+
+    @property
+    def choices(cls):
+        empty = [(None, cls.__empty__)] if hasattr(cls, '__empty__') else []
+        return empty + [(member.value, member.label) for member in cls.__members__.values()]
```

Comment:
I think we can use `in cls` instead of `in cls.__members__.values()` (here and in `names`).